File: android/src/main/java/au/com/crazybean/mobilex/kurir/modules/explore/impl/ExploreModule.kt
package au.com.crazybean.mobilex.kurir.modules.explore.impl

import au.com.crazybean.mobilex.kurir.modules.explore.ExploreDelegate
import au.com.crazybean.mobilex.kurir.modules.explore.ExploreView
import au.com.crazybean.mobilex.kurir.modules.explore.ExploreViewModel
import org.koin.androidx.viewmodel.dsl.viewModel
import org.koin.dsl.module

val exploreModule = module {
    // ViewModel
    viewModel {
        ExploreViewModel(get())
    }

    // Delegate
    factory { (view: ExploreView?) ->
        ExploreDelegate(view, get())
    }
}

File: ios/Kurir/modules/explore/impl/ExploreModule.swift
//
//  ExploreModule.swift
//  Kurir
//
//  Created by Loren on 19/9/19.
//  Copyright © 2019 Crazybean Studio. All rights reserved.
//

import Mobilex

class ExploreModule: Module {
    override func inject() {
        // ViewModel
        factory(ExploreViewModel.self) { r in
            ExploreViewModel(repository: r.resolve(UsersRepository.self))
        }
        
        // Delegate
        factory(ExploreDelegate.self) { (r, view: ExploreView) -> ExploreDelegate in
            ExploreDelegate(view: view, viewModel: r.resolve(ExploreViewModel.self)!)
        }
    }
}
